pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -806,18 +806,16 @@ func (iqr *IQR) AsResult(qType structs.QueryType) (*structs.PipeSearchResponseOu
 			GroupByCols:        aggGroupByCols,
 			// TODO: add IsTimechart flag
 		}
-	case structs.SegmentStatsCmd: // TODO: implement
-		panic("IQR.AsResult: SegmentStatsCmd not implemented")
 	default:
 		return nil, fmt.Errorf("IQR.AsResult: unexpected query type %v", qType)
 	}
 
 	return response, nil
 }
 
-func (iqr *IQR) AppendGroupByResults(bucketHolderArr []*structs.BucketHolder, measureFuncs []string, aggGroupByCols []string, bucketCount int) error {
+func (iqr *IQR) AppendRRCStatsResults(bucketHolderArr []*structs.BucketHolder, measureFuncs []string, aggGroupByCols []string, bucketCount int) error {
```

Comment:
nit: should this be renamed to `AppendStatsResults`? Does it matter that the stats results came from RRCs?

Reply:
It does not I renamed.